File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        map<int, int>cnt;
        for(int num: nums) if(!cnt.contains(num)) cnt[num]++;
        int ans=0;
        for(auto& x:cnt) {
            if(cnt.contains(x.first-1)) {
                x.second = x.second + cnt[x.first-1];
            } 
            ans=max(x.second, ans);
        }
        return ans;
    }
};

File: Data Structures & Algorithms/house-robber/submission-2.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        n = len(nums) -1
        dp = [-1] * (n+2)
        def solve(n: int):
            if n == 0:
                return nums[0]
            elif n == 1:
                return max(nums[0], nums[1])
            if dp[n] != -1:
                return dp[n]
            dp[n] = max(nums[n]+ solve(n-2), solve(n-1))
            return dp[n]
        return solve(n)